### PR TITLE
Docker quickstart

### PR DESCRIPTION
**Pull request by bvberkum**
Had to create a new pull request after force-push for #982 , didn't seem to reopen.

Remade commit with changes this time at requested space in index document. 

Also added a commit updating docker-101 with a bit on beyond pulling the image: creating, access and cleanup.

Hope all the syntax works and is in the right place.

### CODE REVIEW

File: doc/md/docker/docker-101.md [hunk 1]
```diff
@@ -93,3 +98,45 @@ If you care only about trying out docker and not about what is running or saved,
 $ docker rmi -f $(docker images -aq) # remove or mark all images for disposal
 $ docker volume rm $(docker volume ls -q) # remove all volumes
 ```
+
+### SystemD config
```
Comment (nodiscc) [inline doc/md/docker/docker-101.md]: `Systemd` (lowercase D)

File: doc/md/docker/docker-101.md [hunk 2]
```diff
@@ -93,3 +98,45 @@ If you care only about trying out docker and not about what is running or saved,
...
+Systemd is the process manager of choice on ubuntu. Once you have a ``docker`` service installed, you can add use the following steps to set up Shaarli to run on system start.
```
Comment (nodiscc) [inline doc/md/docker/docker-101.md]: s/on ubuntu/on Debian-based distributions/

File: doc/md/docker/docker-101.md [hunk 3]
```diff
@@ -93,3 +98,45 @@ If you care only about trying out docker and not about what is running or saved,
...
+
+# Put any environment you want in here, like $host- or $domainname in this example
+EnvironmentFile=/etc/sysconfig/box-environment
+
+# It's just an example..
+ExecStart=/usr/bin/docker run \
+  -p 28010:80 \
+  --name ${hostname}-shaarli \
+  --hostname shaarli.${domainname} \
+  \
```
Comment (nodiscc) [inline doc/md/docker/docker-101.md]: Unneeded blank line

File: doc/md/docker/docker-101.md [hunk 4]
```diff
@@ -93,3 +98,45 @@ If you care only about trying out docker and not about what is running or saved,
...
+  -v /srv/docker-volumes-local/shaarli-data:/var/www/shaarli/data:rw \
+  -v /etc/localtime:/etc/localtime:ro \
+  \
```
Comment (nodiscc) [inline doc/md/docker/docker-101.md]: Unneeded blank line

File: doc/md/docker/docker-101.md [hunk 5]
```diff
@@ -60,3 +60,83 @@ wheezy: Pulling from debian
...
+Trying out different images can fill some gigabytes of disk quickly. Besides images, the docker volumes usually take up most disk space.
+
+If you care only about trying out docker and not about what is running or saved, the following commands should help you out quickly:
+
+```bash
+$ docker rmi -f $(docker images -aq) # remove or mark all images for disposal
+$ docker volume rm $(docker volume ls -q) # remove all volumes
+```
+
+### SystemD config
```
Comment (nodiscc) [inline doc/md/docker/docker-101.md]: `Systemd` (lowercase D)

File: doc/md/docker/docker-101.md [hunk 6]
```diff
@@ -60,3 +60,83 @@ wheezy: Pulling from debian
...
+Systemd is the process manager of choice on ubuntu. Once you have a ``docker`` service installed, you can add use the following steps to set up Shaarli to run on system start.
```
Comment (nodiscc) [inline doc/md/docker/docker-101.md]: s/on ubuntu/on Debian-based distributions/

File: doc/md/docker/docker-101.md [hunk 7]
```diff
@@ -60,3 +60,83 @@ wheezy: Pulling from debian
 Digest: sha256:c584131da2ac1948aa3e66468a4424b6aea2f33acba7cec0b631bdb56254c4fe
 Status: Downloaded newer image for debian:wheezy
 ```
+
+Docker re-uses layers already downloaded. Iow. if you have images based only Alpine or some Ubuntu version for example, those can share disk space.
```
Comment (nodiscc) [inline doc/md/docker/docker-101.md]: s/Iow/In other words

File: doc/md/docker/docker-101.md [hunk 8]
```diff
@@ -60,3 +60,83 @@ wheezy: Pulling from debian
...
+```
+
+Some containers may be designed or configured to be restarted, others are not. Note that both network ports and volumes of a container are created on start, and not editable later.
+
+### Access a running container
+A running container is accessible using ``docker exec``, or ``docker copy``. You can use ``exec`` to start a root shell in the Shaarli container:
+```bash
+$ docker exec -ti <container-name-or-id> bash
+```
+Note the names and ID's of containers are list in ``docker ps``. You an even type only one or two letters of the ID, given they are unique.
```
Comment (nodiscc) [inline doc/md/docker/docker-101.md]: s/are list/are listed

Comment (nodiscc) [reply doc/md/docker/docker-101.md]: s/you an/you can

**Comment by virtualtam** [comment]
Thanks @bvberkum for updating the PR :)

**Pull request merged**